hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/metrics/RBFMetrics.java

```diff
@@ -544,28 +548,30 @@ public String getNodeUsage() {
 
     final Map<String, Map<String, Object>> info = new HashMap<>();
     try {
-      RouterRpcServer rpcServer = this.router.getRpcServer();
-      DatanodeInfo[] live = rpcServer.getDatanodeReport(
-          DatanodeReportType.LIVE, false, timeOut);
-
-      if (live.length > 0) {
-        float totalDfsUsed = 0;
-        float[] usages = new float[live.length];
-        int i = 0;
-        for (DatanodeInfo dn : live) {
-          usages[i++] = dn.getDfsUsedPercent();
-          totalDfsUsed += dn.getDfsUsedPercent();
-        }
-        totalDfsUsed /= live.length;
-        Arrays.sort(usages);
-        median = usages[usages.length / 2];
-        max = usages[usages.length - 1];
-        min = usages[0];
-
-        for (i = 0; i < usages.length; i++) {
-          dev += (usages[i] - totalDfsUsed) * (usages[i] - totalDfsUsed);
+      if (this.enableGetDNUsage) {
+        RouterRpcServer rpcServer = this.router.getRpcServer();
+        DatanodeInfo[] live = rpcServer.getDatanodeReport(
+            DatanodeReportType.LIVE, false, timeOut);
+
+        if (live.length > 0) {
+          float totalDfsUsed = 0;
+          float[] usages = new float[live.length];
+          int i = 0;
+          for (DatanodeInfo dn : live) {
+            usages[i++] = dn.getDfsUsedPercent();
+            totalDfsUsed += dn.getDfsUsedPercent();
+          }
+          totalDfsUsed /= live.length;
+          Arrays.sort(usages);
+          median = usages[usages.length / 2];
+          max = usages[usages.length - 1];
+          min = usages[0];
+
+          for (i = 0; i < usages.length; i++) {
+            dev += (usages[i] - totalDfsUsed) * (usages[i] - totalDfsUsed);
+          }
+          dev = (float) Math.sqrt(dev / usages.length);
```

Comment:
Apache commons math has a nice StandardDeviation utility.
It might be good to use this directly.
https://commons.apache.org/proper/commons-math/javadocs/api-3.6/org/apache/commons/math3/stat/descriptive/moment/StandardDeviation.html